pkg/cosign/verify.go

```diff
@@ -291,6 +291,88 @@ func ValidateAndUnpackCert(cert *x509.Certificate, co *CheckOpts) (signature.Ver
 	return verifier, nil
 }
 
+// ValidateAndUnpackCertWithIntermediates function has same logic as ValidateAndUnpackCert.
+// Main diffrence is that we can not use ValidateAndUnpackCert in multi thread invocation
+// as it uses a common reference of CheckOpts so in multi thread invocation co.intermediateCert object
+// can get overwritten my multiple threads. So to solve this ValidateAndUnpackCertWithIntermediates
+// is created which has a separate argument for Intermediate certs.
+func ValidateAndUnpackCertWithIntermediates(cert *x509.Certificate, co *CheckOpts, intermediateCerts *x509.CertPool) (signature.Verifier, error) {
```

Comment:
To avoid duplicated code, can we call `ValidateAndUnpackCertWithIntermediates` from `ValidateAndUnpackCert`? Would change `ValidateAndUnpackCert` to just `return ValidateAndUnpackCertWithIntermediates(cert, co, co.IntermediateCerts)`

Reply:
Agreed i will incorporate the review comment